website/utils/url_utils.ts

```diff
@@ -955,8 +960,12 @@ export async function parseViewerUrlParams(urlSearchParams: URLSearchParams): Pr
 
   // Parse data sources (URL or dataset/id pair)
   if (params.url) {
-    const imageUrls = tryDecodeURLList(params.url) ?? decodeURL(params.url);
-    const firstUrl = Array.isArray(imageUrls) ? imageUrls[0] : imageUrls;
+    const sceneUrls = tryDecodeURLList(params.url, " ") ?? [params.url];
+    const sourceUrls = sceneUrls.map((scene) => tryDecodeURLList(scene) ?? [decodeURL(scene)]);
+
+    const firstScene = sourceUrls[0];
+    const firstUrl = Array.isArray(firstScene) ? firstScene[0] : firstScene;
+    const imageUrls = sourceUrls.length === 1 ? (sourceUrls[0].length === 1 ? firstScene[0] : firstScene) : sourceUrls;
```

Comment:
it has become too hard to parse these 5 lines. I think types on the variable declarations would help to understand.  And maybe rename imageUrls and firstUrl to be imageUrlsOfFirstScene and firstUrlOfFirstScene?  if that's even accurate?

Reply:
My little refactor didn't make this much better, but I did add some docs and a type signature. Let me know if there's more I can do?